clients/src/main/java/org/apache/kafka/common/Metric.java

```diff
@@ -27,8 +27,16 @@
     public MetricName metricName();
 
     /**
-     * The value of the metric
+     * The value of the metric as double if the metric is measurable
+     * @throws IllegalStateException if this metric does not have a measurable double value
```

Comment:
Should probably add a note to upgrade.html stating that this is possible for users who iterate over the metrics returned by clients like `KafkaProducer` and `KafkaConsumer`.